TLM/TLM/Patch/_CitizenAI/GetPathTargetPositionPatch.cs

```diff
@@ -0,0 +1,34 @@
+namespace TrafficManager.Patch._CitizenAI {
+    using HarmonyLib;
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+    using TrafficManager.Util.Extensions;
+
+    [HarmonyPatch(typeof(CitizenAI), "GetPathTargetPosition")]
+    public static class GetPathTargetPositionPatch {
+        [HarmonyPriority(Priority.Low)]
+        public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions) {
+            var mGetGap_ =
+                AccessTools.Method(typeof(GetPathTargetPositionPatch), nameof(GetPathTargetPositionPatch.GetGap));
+            foreach (var instruction in instructions) {
+                yield return instruction;
+                if (instruction.opcode == OpCodes.Ldc_R4 && instruction.operand is float value && value == 64) {
+                    // we don't replace LDC_R4 64 but rather pass it to GetGap. This way it would be more compatible with other mods in future should they try to modify the same line.
+                    yield return new CodeInstruction(OpCodes.Ldloc, 4); // path position
+                    yield return new CodeInstruction(OpCodes.Call, mGetGap_);
+                }
+            }
+        }
+
+        private static float GetGap(float gap /*64f*/, PathUnit.Position pathPos) {
+            NetInfo info = pathPos.m_segment.ToSegment().Info;
+            return Max(gap, info.m_minCornerOffset, info.m_halfWidth * 2);
+        }
+
+        private static float Max(float a, float b, float c) {
+            float max = a > b ? a : b;
+            max = max > c ? max : c;
+            return max;
```

Comment:
I think you can join those two lines

Reply:
yeah I know. I can also avoid using extra max variable. 
doesn't the compiler do that in release build though? I thought this is more readable

Reply:
compiler does it this way
```cs
private static float Max(float a, float b, float c)
{
	float num = (a > b) ? a : b;
	return (num > c) ? num : c;
}
```
So it does not try to reduce number of variables.

Reply:
I'm not sure how you would avoid using extra max variable in this case. What you see it's generated IL, what is going on after Jit no one knows.
I would just merge the return and that's all.

Reply:
Will it compile to better code if you express as `Math.Max(Math.Max(a, b), c)`?

Reply:
I doubt, at 99% method call won't be inlined. It's simple code, no need to overthink it 😉

Reply:
In fact manual comparing and return is shorter. Is it faster? I don't know. Godbolt compiler explorer added some C#

![bild](https://user-images.githubusercontent.com/288724/178774253-dd78f3ce-e6dd-4c9b-bde7-91fc541c666c.png)
![bild](https://user-images.githubusercontent.com/288724/178774295-7203004c-7319-4b71-9acc-962852282c7c.png)
![bild](https://user-images.githubusercontent.com/288724/178774465-a2bdb1db-4eb8-44ac-8403-cf1ee07f4cb1.png)

Reply:
Take it with grain of salt, even our benchmark project gives incorrect performance results compared to similar benchmark run inside the game (benchmark project code is much faster than running in the game because inlining is not as aggressive as it could be).